docs/security_self_assessment.md

```diff
@@ -0,0 +1,1588 @@
+
+Where control audits differ from the original CIS benchmark, the audit commands specific to RKE2 are provided for testing. When performing the tests, you will need access to the command line on the hosts of the RKE2 nodes. The commands also make use of the the `jq` and `kubectl` (with valid config) tools to and are required in the testing and evaluation of test results.
+
+## Controls
+
+### 1 Master Node Security Configuration
+
+#### 1.1 Master Node Configuration Files
+
+*1.1.1* Ensure that the API server pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the API server. All configuration is passed in as arguments at container run time.
+
+*1.1.2* Ensure that the API server pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the API server. All configuration is passed in as arguments at container run time.
+
+*1.1.3* Ensure that the controller manager pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the controller manager. All configuration is passed in as arguments at container run time.
+
+*1.1.4* Ensure that the controller manager pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the controller manager. All configuration is passed in as arguments at container run time.
+
+*1.1.5* Ensure that the scheduler pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the scheduler. All configuration is passed in as arguments at container run time.
+
+*1.1.6* Ensure that the scheduler pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the scheduler. All configuration is passed in as arguments at container run time.
+
+*1.1.7* Ensure that the etcd pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for etcd. All configuration is passed in as arguments at container run time.
+
+*1.1.8* Ensure that the etcd pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for etcd. All configuration is passed in as arguments at container run time.
+
+*1.1.11* Ensure that the etcd data directory permissions are set to `700` or more restrictive (Scored)
+
+Result: Not Applicable
```

Comment:
Why would this be not applicable? The etcd data directory is at `/var/lib/rancher/rke2/server/db/etcd` and should have permissions set to `700`.

I can see this is currently not the case with rke2 and it was likely lost in the script refactor and never picked up elsewhere.